editoast/src/views/infra/objects.rs

```diff
@@ -0,0 +1,99 @@
+use crate::api_error::ApiError;
+use crate::infra::Infra;
+use crate::{api_error::ApiResult, db_connection::DBConnection, schema::ObjectType};
+use diesel::sql_types::Text;
+use diesel::{sql_query, RunQueryDsl};
+use rocket::http::Status;
+use rocket::response::status::Custom;
+use rocket::serde::json::{json, Error as JsonError, Json, Value as JsonValue};
+
+#[derive(Debug, thiserror::Error)]
+enum GetObjectsErrors {
+    #[error("No object ids provided")]
+    NoObjectIdsProvided,
+}
+
+impl ApiError for GetObjectsErrors {
+    fn get_status(&self) -> Status {
+        match self {
+            GetObjectsErrors::NoObjectIdsProvided => Status::BadRequest,
+        }
+    }
+
+    fn get_type(&self) -> &'static str {
+        match self {
+            GetObjectsErrors::NoObjectIdsProvided => "editoast:infra:objects:NoObjectIdsProvided",
+        }
+    }
+}
+
+/// Return the railjson list of a specific OSRD object
+#[get("/<infra_id>/objects/<object_type>", data = "<obj_ids>")]
+pub async fn get_objects<'a>(
+    infra_id: i32,
+    object_type: ObjectType,
+    obj_ids: Result<Json<Vec<String>>, JsonError<'a>>,
+    conn: DBConnection,
+) -> ApiResult<Custom<JsonValue>> {
+    let obj_ids = obj_ids?;
+    if obj_ids.is_empty() {
+        return Err(GetObjectsErrors::NoObjectIdsProvided.into());
+    }
+    let obj_ids = obj_ids.into_inner();
+    conn.run(move |conn| {
+        let infra = Infra::retrieve(conn, infra_id).unwrap();
+        sql_query(format!(
+            "SELECT $1.obj_id,
+            $1.data,
+            ST_AsGeoJSON(geographic) as geo,
+            ST_AsGeoJSON(schematic) as sch
+        FROM osrd_infra_$1layer
+            JOIN $1 ON $2.obj_id = $1.obj_id
+        WHERE $1.obj_id IN (
+                'rt.DE7->DD4',
+                'rt.DG0->DG1_1',
+                'rt.DD1_8->DD1_7'
+            ) AND $1.id = {}",
+            infra.id,
```

Comment:
You can not  bind table name. You need to use `format!` for tables. However for values such as `infra_id` or `obj_ids` you **MUST** use bindings